### PR TITLE
Add Haxe key_features

### PR DESCRIPTION
**Pull request by mfoda**
@ErikSchierboom I took some liberty with the "icon" fields since the list of icons is not yet finalized. Let me know if these can be improved.

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -295,6 +295,37 @@
...
+    },
+    {
+      "icon": "features-general-purpose",
+      "title": "General purpose",
+      "content": "Haxe excels in game development but is used in many domains, including web, mobile, and desktop."
+    },
+    {
+      "icon": "features-native",
+      "title": "Native access",
+      "content": "Haxe externs enable type-safe interaction with the underlying target, providing access to native APIs."
```
Comment (ErikSchierboom) [inline config.json:322]: This field exceeds the maximum length of 100 by just a bit. See [the docs](https://github.com/exercism/docs/blob/main/anatomy/tracks/config-json.md#key-features).

**Review by ErikSchierboom — COMMENTED** [review]
One small nit.

File: config.json (hunk 1 above)
Comment (mfoda) [reply config.json:322]: Ah, last minute wording change. I missed that :-)

File: config.json [hunk 2]
```diff
@@ -295,6 +295,37 @@
...
+      "title": "Native access",
+      "content": "Externs enable type-safe interaction with the underlying target, providing access to native APIs."
+    },
+    {
+      "icon": "features-metaprogramming",
+      "title": "Metaprogramming",
+      "content": "Syntax-transformation using macros unlocks the power to fit the language to your problem domain."
+    }
+  ],
   "tags": []
```
Comment (ErikSchierboom) [inline config.json:330]: Does this overwrite the tags again?

**Review by ErikSchierboom — APPROVED** [review]

File: config.json (hunk 2 above)
Comment (mfoda) [reply config.json:330]: My bad, seems like it will also remove `concepts`. Should I make a commit that puts everything together back in place?

**Pull request merged**